### ISSUE TITLE
Few-shot results on EuroSAT dataset

### ISSUE THREAD

**Issue by wonjeongchoi** [lionel-w2/FAP#3]
Hi, 
thanks for your nice work 👍 

Could you tell me how did you set $\lambda$ to make results in the paper?

I know that the provided sh files set $\lambda$ to 1.5 for few-shot setting on all downstream datasets, but when I run EuroSAT with $\lambda$ of 1.5, its adversarial accuracy is almost close to zero. (I attach a portion of the log file below)

> ...
Initialize tensorboard (log_dir=./output/few_shot/adv_term-cos_eps-1_alpha-1_train_iter-2_test_iter-100_lambda1-1.5/FAP_vit_b32_ep10_batch4_2ctx_notransform/eurosat_shot16/seed1/tensorboard)
epoch [1/10] batch [20/40] time 0.590 (0.710) data 0.000 (0.020) loss 3.6523 (6.0036) Nat_loss 1.4316 (1.5562) Adv_loss 1.4805 (2.9647) lr 1.0000e-05 eta 0:04:29
epoch [1/10] batch [40/40] time 0.601 (0.642) data 0.000 (0.010) loss 2.9648 (5.0879) Nat_loss 2.1992 (1.6938) Adv_loss 0.5103 (2.2629) lr 3.5000e-03 eta 0:03:51
epoch [2/10] batch [20/40] time 0.550 (0.660) data 0.000 (0.022) loss 2.2969 (2.3324) Nat_loss 2.2969 (2.2759) Adv_loss 0.0001 (0.0377) lr 3.5000e-03 eta 0:03:44
epoch [2/10] batch [40/40] time 0.774 (0.697) data 0.000 (0.011) loss 2.2871 (2.3127) Nat_loss 2.2812 (2.2832) Adv_loss 0.0035 (0.0197) lr 3.4143e-03 eta 0:03:42
epoch [3/10] batch [20/40] time 1.172 (1.001) data 0.001 (0.022) loss 2.2754 (2.2859) Nat_loss 2.2715 (2.2836) Adv_loss 0.0023 (0.0015) lr 3.4143e-03 eta 0:05:00
epoch [3/10] batch [40/40] time 1.223 (1.066) data 0.000 (0.011) loss 2.2773 (2.2875) Nat_loss 2.2734 (2.2852) Adv_loss 0.0027 (0.0016) lr 3.1658e-03 eta 0:04:58
epoch [4/10] batch [20/40] time 1.637 (1.533) data 0.001 (0.023) loss 2.2949 (2.2861) Nat_loss 2.2930 (2.2833) Adv_loss 0.0012 (0.0020) lr 3.1658e-03 eta 0:06:38
epoch [4/10] batch [40/40] time 1.579 (1.517) data 0.000 (0.012) loss 2.3027 (2.2843) Nat_loss 2.2988 (2.2814) Adv_loss 0.0028 (0.0019) lr 2.7786e-03 eta 0:06:04
epoch [5/10] batch [20/40] time 1.465 (1.541) data 0.000 (0.023) loss 2.2734 (2.2782) Nat_loss 2.2695 (2.2745) Adv_loss 0.0026 (0.0024) lr 2.7786e-03 eta 0:05:39
epoch [5/10] batch [40/40] time 1.510 (1.516) data 0.000 (0.012) loss 2.2676 (2.2776) Nat_loss 2.2656 (2.2738) Adv_loss 0.0012 (0.0025) lr 2.2908e-03 eta 0:05:03
epoch [6/10] batch [20/40] time 1.624 (1.534) data 0.000 (0.028) loss 2.2676 (2.2672) Nat_loss 2.2637 (2.2635) Adv_loss 0.0020 (0.0026) lr 2.2908e-03 eta 0:04:36
epoch [6/10] batch [40/40] time 1.484 (1.505) data 0.000 (0.014) loss 2.2930 (2.2672) Nat_loss 2.2910 (2.2631) Adv_loss 0.0019 (0.0028) lr 1.7500e-03 eta 0:04:00
epoch [7/10] batch [20/40] time 1.414 (1.537) data 0.000 (0.023) loss 2.2852 (2.2604) Nat_loss 2.2773 (2.2541) Adv_loss 0.0047 (0.0040) lr 1.7500e-03 eta 0:03:35
epoch [7/10] batch [40/40] time 1.518 (1.528) data 0.000 (0.012) loss 2.3438 (2.2571) Nat_loss 2.3242 (2.2491) Adv_loss 0.0125 (0.0053) lr 1.2092e-03 eta 0:03:03
epoch [8/10] batch [20/40] time 1.538 (1.544) data 0.000 (0.024) loss 2.2305 (2.2356) Nat_loss 2.2188 (2.2185) Adv_loss 0.0080 (0.0114) lr 1.2092e-03 eta 0:02:34
epoch [8/10] batch [40/40] time 1.604 (1.523) data 0.000 (0.012) loss 2.3066 (2.2422) Nat_loss 2.2969 (2.2228) Adv_loss 0.0069 (0.0130) lr 7.2138e-04 eta 0:02:01
epoch [9/10] batch [20/40] time 1.408 (1.521) data 0.000 (0.023) loss 2.2461 (2.2276) Nat_loss 2.2090 (2.2005) Adv_loss 0.0248 (0.0183) lr 7.2138e-04 eta 0:01:31
epoch [9/10] batch [40/40] time 1.587 (1.511) data 0.000 (0.012) loss 2.2500 (2.2237) Nat_loss 2.2266 (2.1978) Adv_loss 0.0152 (0.0173) lr 3.3422e-04 eta 0:01:00
epoch [10/10] batch [20/40] time 1.497 (1.548) data 0.000 (0.021) loss 2.2988 (2.2232) Nat_loss 2.2676 (2.1966) Adv_loss 0.0205 (0.0177) lr 3.3422e-04 eta 0:00:30
epoch [10/10] batch [40/40] time 1.501 (1.528) data 0.000 (0.011) loss 2.1602 (2.2104) Nat_loss 2.1484 (2.1847) Adv_loss 0.0084 (0.0171) lr 8.5651e-05 eta 0:00:00
Checkpoint saved to ./output/few_shot/adv_term-cos_eps-1_alpha-1_train_iter-2_test_iter-100_lambda1-1.5/FAP_vit_b32_ep10_batch4_2ctx_notransform/eurosat_shot16/seed1/MultiModalPromptLearner/model.pth.tar-10
Finish training
Deploy the last-epoch model
Loading evaluator: Classification
>Evaluate on the *test* set --Adversary
>
>=> result
>
>* total: 8,100
>* correct: 3,381
>* accuracy: 41.7%
>* error: 58.3%
>* macro_f1: 35.0%
>
>=> result
>
>* total: 8,100
>* correct: 15
>* accuracy: 0.2%
>* error: 99.8%
>* macro_f1: 0.2%
>Elapsed: 0:48:58

Since the results on the other datasets seems similar to the paper, I guess some hyperparameters like $\lambda$ is wrong for EuroSAT dataset.
Could you tell me what is wrong with this and how to set tthe hyperparameters like $\lambda$?

Thank you again 😄

**Comment by SensenGao** [comment]
I found the same situation

**Comment by wonjeongchoi** [comment]
Hi, is there any update on my issue?
It would be very helpful to improve your work and research society.
